ssl/statem/extensions_clnt.c

```diff
@@ -198,22 +198,33 @@ EXT_RETURN tls_construct_ctos_supported_groups(SSL *s, WPACKET *pkt,
         uint16_t ctmp = pgroups[i];
 
         if (tls_curve_allowed(s, ctmp, SSL_SECOP_CURVE_SUPPORTED)) {
-            if (!WPACKET_put_bytes_u16(pkt, ctmp)) {
-                    SSLfatal(s, SSL_AD_INTERNAL_ERROR,
-                             SSL_F_TLS_CONSTRUCT_CTOS_SUPPORTED_GROUPS,
-                             ERR_R_INTERNAL_ERROR);
-                    return EXT_RETURN_FAIL;
-                }
+#ifndef OPENSSL_NO_TLS1_3
+            int ctmp13 = ssl_group_id_internal_to_tls13(ctmp);
+            if (ctmp13 != 0 && ctmp13 != ctmp) {
+                int ver_min, ver_max;
+                if (ssl_get_min_max_version (s, &ver_min, &ver_max, NULL))
```

Comment:
Nit: spurious  space after function name

Reply:
Can we move this function call to before the "for" loop - so we don't have to call it every time? The ver_min and ver max values won't change.

Reply:
The function is only rarely needed, I was kind of trying not to add overhead to
the normal case, which is no brainpool curves.  These are only meant as
a fallback for people who do not trust in the NIST curves, and might like
to have a different fallback instead.  So I think that is at most called 3 times,
once for each brainpool curve NID.
But I can do that if it is considered too ugly.

Reply:
Additionally it is an error path that is not needed when no brainpool
curve is to be enabled.